core/src/main/java/com/amplifyframework/core/model/SerializedModel.java

```diff
@@ -206,7 +206,7 @@ public static SerializedModel merge(SerializedModel incoming, SerializedModel ex
 
     @NonNull
     @Override
-    public String getId() {
+    public String resolveIdentifier() {
```

Comment:
Should the return type of this method be `Serializable` as it maybe a subclass of `ModelPrimaryKey` for models using custom PK + SKs?

Reply:
Done.